pkg/storage/apis/app/v1beta2/fybrikstorageaccount_types.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta2
+
+import (
+	"encoding/json"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"fybrik.io/fybrik/pkg/model/taxonomy"
+	"fybrik.io/fybrik/pkg/serde"
+)
+
+const typeKey = "type"
+const idKey = "id"
+const geographyKey = "geography"
+const regionKey = "region"
+const secretRefKey = "secretRef"
+
+// FybrikStorageAccountSpec defines the desired state of FybrikStorageAccount
+// +kubebuilder:pruning:PreserveUnknownFields
+type FybrikStorageAccountSpec struct {
+	// Identification of a storage account
+	// +required
+	ID string `json:"id"`
+	// +required
+	// A name of k8s secret deployed in the control plane.
+	SecretRef string `json:"secretRef"`
+	// +required
+	// Type of the storage, e.g., s3
+	Type taxonomy.ConnectionType `json:"type"`
+	// +required
+	// Storage geography
+	Geography taxonomy.ProcessingLocation `json:"geography"`
+	// Additional storage properties, specific to the storage type
+	AdditionalProperties serde.Properties `json:"-"`
+}
+
+// FybrikStorageAccountStatus defines the observed state of FybrikStorageAccount
+type FybrikStorageAccountStatus struct {
+}
+
+// FybrikStorageAccount defines a storage account used for copying data.
```

Comment:
Perhaps a better description would be ...
FybrikStorageAccount is a storage account Fybrik uses to dynamically allocate space for datasets whose creation or copy it is orchestrates

Reply:
Done